src/Service/Program.cs

```diff
@@ -195,6 +195,21 @@ public static ILoggerFactory GetLoggerFactoryForLogLevel(LogLevel logLevel, Tele
                         });
                     }
 
+                    if (Startup.AzureLogAnalyticsOptions.Enabled && !string.IsNullOrWhiteSpace(Startup.AzureLogAnalyticsOptions.Auth?.CustomTableName) &&
+                        !string.IsNullOrWhiteSpace(Startup.AzureLogAnalyticsOptions.Auth?.DcrImmutableId) && !string.IsNullOrWhiteSpace(Startup.AzureLogAnalyticsOptions.Auth?.DceEndpoint))
```

Comment:
We can move all these checks to check if AzureLogAnalytics is enabled or not into a helper function and reuse it in Startup to create the singleton AzureFlusherService.